rust/processor/src/processors/token_v2_processor.rs

```diff
@@ -770,41 +771,66 @@ async fn parse_v2_token(
+                } else {
+                    if let Some(old_burn_event) = BurnEvent::from_event(event, txn_version).unwrap()
                     {
-                        // we don't want index to be 0 otherwise we might have collision with write set change index
-                        // note that these will be multiplied by -1 so that it doesn't conflict with wsc index
-                        let index = if index == 0 {
-                            user_txn.events.len()
-                        } else {
-                            index
-                        };
-                        aggregated_data
-                            .transfer_events
-                            .push((index as i64, transfer_events));
+                        if let Some(("burn", token_address)) = &previous_mint_or_burn_event {
+                            if token_address == &old_burn_event.get_token_address() {
+                                continue;
+                            }
+                        }
+                        let burn_event = Burn::new(
+                            standardize_address(
+                                event.key.as_ref().unwrap().account_address.as_str(),
+                            ),
+                            old_burn_event.get_token_address(),
+                            "".to_string(),
+                        );
+                        tokens_burned.insert(burn_event.get_token_address(), burn_event);
+                    } else if let Some(mint_event) =
+                        MintEvent::from_event(event, txn_version).unwrap()
+                    {
+                        if let Some(("mint", token_address)) = &previous_mint_or_burn_event {
+                            if token_address == &mint_event.get_token_address() {
+                                continue;
+                            }
+                        }
+                        tokens_minted.insert(mint_event.get_token_address());
+                    } else if let Some(transfer_events) =
+                        TransferEvent::from_event(event, txn_version).unwrap()
+                    {
+                        if let Some(aggregated_data) =
+                            token_v2_metadata_helper.get_mut(&transfer_events.get_object_address())
+                        {
+                            // we don't want index to be 0 otherwise we might have collision with write set change index
+                            // note that these will be multiplied by -1 so that it doesn't conflict with wsc index
+                            let index = if index == 0 {
+                                user_txn.events.len()
+                            } else {
+                                index
+                            };
+                            aggregated_data
+                                .transfer_events
+                                .push((index as i64, transfer_events));
+                        }
```

Comment:
Why is this changed so much? I thought `should_skip` handles skipping the v1 events

Reply:
...Because... Igor makes them mixed now. go to the smart contracts and then you will know why. That's why I want to separate a fix for kent first.

Reply:
Chatted offline about this. There could be v1 and v2 event next to each other for different collections and could accidentally skip a v1 event 

To get around this, we'll create `should_skip` that only skips if event names AND collection address matches.